### ISSUE TITLE
Restrict to Specific Products?

### ISSUE THREAD

**Issue by thollis78** [Prospress/woocommerce-subscriptions-gifting#193]
I have multiple subscriptions but only one is "giftable". Is there a way to restrict the gifting to specific products?

**Comment by james-allan** [comment]
Hi @thollis78 

> Is there a way to restrict the gifting to specific products?

There isn't a way to do it through a UI but you can restrict gifting to specific products with custom code using the `wcsg_is_giftable_product` filter. 

Something like this: 

```php
function eg_is_giftable_product( $is_giftable, $product ) {
    $giftable_product_ids = array( 24, 12859, 11418 );

    return in_array( $product->id , $giftable_product_ids );
}
add_filter( 'wcsg_is_giftable_product', 'eg_is_giftable_product', 10, 2 );
```

This uses a product whitelist where all the products you want to display the "This is a gift" option should be included in the `$giftable_product_ids` array. All other products will not display the gift option.

**Comment by tfarrokhnia** [comment]
I understand how this works in theory, but where would I add this code?

**Comment by Nat-THS** [comment]
@tfarrokhnia Normally, in your child theme's functions.php

**Comment by tfarrokhnia** [comment]
@Nat-THS Yes, I had thought that but when I tried originally it broke everything. Turned out it was an FTP issue and not the new code. Everything working just fine now! Thanks!

**Comment by SJCGraphics** [comment]
I Hi. I used the code above and it is still showing up. 

I deleted this line here from the plugin to remove it from the single product at all so they could only buy the gift bundle and not all products as gifts: 

		add_action( 'woocommerce_before_add_to_cart_button', __CLASS__ . '::add_gifting_option_product' );

When they go to checkout however from stork-bites.com/shop, the checkbox still shows up for all the products. I only want it to show for product ID 2064. Thanks. Any insight appreciated as this code is not working for me. I need the checkbox only displaying for one product. Thanks.

**Comment by james-allan** [comment]
Hi @SJCGraphics 

Sorry for the delay in getting back to you. Your message has slipped through, sorry.

Could you let me know if you're still having issues? If you are, please send me the code you're using and let me know what version of WooCommerce and WC Subscriptions you are running.
